lib/smart_proxy_dynflow_core/log.rb

```diff
@@ -54,13 +54,28 @@ def roll_log
       end
     end
 
+    class ProxyStructuredFormater < ::Dynflow::LoggerAdapters::Formatters::Abstract
+      def call(_severity, _datetime, _prog_name, message)
+        if ::Exception === message
+          @base.exception("Error details", message)
+          "#{message.message} (#{message.class})"
+        else
+          message
+        end
```

Comment:
Basically we want to use helper called `exception` which is similar to what we have in Foreman core. It reports exteptions via WARNING level, adds properly formatted stacktrace and it also logs it via structured fields (exception class, message, stacktrace). Then we return the textual form (without joining backtrace).